buildenv/jenkins/JenkinsfileBase

```diff
@@ -141,6 +141,15 @@ def setupEnv() {
 	} else {
 		sh 'printenv'
 	}
+        if ( params.USE_GIT_SSH ) {
+                def gitGlobalConfig = fileExists('~/.gitconfig') ? sh(script:"git config --global -l", returnStdout:true) : ''
+                if (!gitGlobalConfig || !gitGlobalConfig.contains('url.git@github.com:.insteadof')) {
```

Comment:
I think overwriting should be fine, no need to check for this condition. Or even better, do `git config --local ...` to install the preferences on the current cloned directory (Jenkins workspace for the current build)

Local (repository) level configs take precedence over global ones.